### PR TITLE
add kube-registry-proxy chart to incubator

### LINKED ISSUE #285: Add chart for kube-registry-proxy
We use helm quite extensively @ Deis (surprise, surprise). It would be cool if there was a chart available for [kube-registry-proxy](https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/registry) that other packages could declare as an external dependency.

At the moment we just maintain our own chart and supply the relevant templates/values and reference the kube-registry-proxy image in the daemonset manifest. If we could add an external chart as part of the meta-charts' [requirements.yaml](https://github.com/kubernetes/helm/blob/935779157bb38a954379d1d11e8b053131ea6c06/docs/helm/helm_dependency.md), it would reduce some of the packaging burden.

I will be more than happy to procure such a chart. I'm just filing this as a reminder for myself. :)

### PR DESCRIPTION
**Pull request by bacongobbler**
closes #285

### CODE REVIEW

**Comment by viglesiasce** [comment]
@k8s-bot ok to test

**Comment by viglesiasce** [comment]
Hey @bacongobbler!

Looks like this is a malformed chart. Its missing the _helpers file that provides some of the templates you are using. Its easiest to just use `helm create` then dump your templates into the templates folder. Also dont forget to add a NOTES.txt file in your templates folder that instructs user on how to get started.

**Comment by bacongobbler** [comment]
thanks @viglesiasce. I have applied those changes :)

File: incubator/kube-registry-proxy/values.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+image:
+  repository: gcr.io/google_containers/kube-registry-proxy
+  tag: 0.4
```
Comment (viglesiasce) [inline incubator/kube-registry-proxy/values.yaml:3]: This tag was not found for the repo:
```console
  52s   12s     3       {kubelet gke-junkyard-default-pool-0ca72bdc-botz}                                         Warning  FailedSync      Error syncing pod, skipping: failed to "StartContainer" for "kube-registry-proxy" with ErrImagePull: "Tag 0.4 not found in repository gcr.io/google_containers/kube-registry-proxy"
```

**Review by viglesiasce — CHANGES_REQUESTED** [review]
Please add a NOTES.txt file explaining how to use this chart. Also please fix tag version as the chart does not currently install properly.

Can you also move this to the stable folder?

File: incubator/kube-registry-proxy/values.yaml (hunk 1 above)
Comment (bacongobbler) [reply incubator/kube-registry-proxy/values.yaml:3]: Yeah, someone from the k8s maintainer team has got to push the new release (pinging @roberthbailey from https://github.com/kubernetes/kubernetes/pull/35797)

Comment (bacongobbler) [reply incubator/kube-registry-proxy/values.yaml:3]: Filed https://github.com/kubernetes/kubernetes/issues/38843 for posterity

Comment (bacongobbler) [reply incubator/kube-registry-proxy/values.yaml:3]: This image has been finally pushed to gcr.io:

```
><> d pull gcr.io/google_containers/kube-registry-proxy:0.4
0.4: Pulling from google_containers/kube-registry-proxy
5040bd298390: Pull complete 
d7a91cdb22f0: Pull complete 
9cac4850e5df: Pull complete 
decd9736c768: Pull complete 
2bb4bfc281bb: Pull complete 
Digest: sha256:1040f25a5273de0d72c54865a8efd47e3292de9fb8e5353e3fa76736b854f2da
Status: Downloaded newer image for gcr.io/google_containers/kube-registry-proxy:0.4
```

**Comment by bacongobbler** [comment]
thanks for the review @viglesiasce. I'll get back to addressing your comments some time early this week now that kube-registry-proxy:0.4 has been pushed out.

**Review by viglesiasce — APPROVED** [review]

File: incubator/kube-registry-proxy/templates/daemon.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      - name: kube-registry-proxy
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        env:
+        - name: REGISTRY_HOST
+          value: {{ .Values.registry.host | quote }}
+        - name: REGISTRY_PORT
+          value: {{ .Values.registry.port | quote }}
+        ports:
+        - containerPort: 80
```
Comment (lachie83) [inline incubator/kube-registry-proxy/templates/daemon.yaml:26]: Should the port be configurable?

Comment (bacongobbler) [reply incubator/kube-registry-proxy/templates/daemon.yaml:26]: You'd have to refactor the underlying image to change it, so I don't think there's any use in making it configurable.

Comment (lachie83) [reply incubator/kube-registry-proxy/templates/daemon.yaml:26]: No worries.

**Comment by lachie83** [comment]
Should we specific a default host? The UX if you don't is the container crashes because `REGISTRY_HOST` is not defined.

**Comment by lachie83** [comment]
Set default registry.host to "gcr.io"

**Comment by lachie83** [comment]
LGTM

**Pull request merged**